include/envoy/upstream/upstream.h

```diff
@@ -301,6 +302,12 @@ class Cluster : public virtual HostSet {
   virtual ClusterInfoConstSharedPtr info() const PURE;
 
   /**
+   * @return a shared pointer to the cluster's outlier detector. If an outlier detector has not been
+   *         installed, returns a nullptr.
+   */
+  virtual const Outlier::DetectorSharedPtr outlierDetector() const PURE;
```

Comment:
return const shared pointer has no meaning. You need to return a pointer to a const Detector.